sensor/kubernetes/eventpipeline/pipeline_impl.go

```diff
@@ -102,13 +122,15 @@ func (p *eventPipeline) Notify(event common.SensorComponentEvent) {
 		// Start listening to events if not yet listening
 		if p.offlineMode.CompareAndSwap(true, false) {
 			log.Infof("Connection established: Starting Kubernetes listener")
-			if err := p.listener.Start(); err != nil {
+			p.createNewContext()
+			if err := p.listener.StartWithContext(p.context); err != nil {
 				log.Fatalf("Failed to start listener component. Sensor cannot run without listening to Kubernetes events: %s", err)
 			}
 		}
 	case common.SensorComponentEventOfflineMode:
 		// Stop listening to events
 		if p.offlineMode.CompareAndSwap(false, true) {
+			p.stopCurrentContext()
```

Comment:
Are `createNewContext` and `stopCurrentContext` going to be changed after the `contextProvider` is merged? If yes, can we link the ticket to change this later?

Reply:
Done